### PR TITLE
External Review for Back-end API

### PR DESCRIPTION
**Pull request by SaveryIV**
In this project, the following was implemented:

- We separately initialized a rails repository for the backend
- We created models for Doctors, users and reservations
- We created controllers with actions for the various models
- We setup API documentation for our backend.
- Implemented unit testing on our implementations
- We deployed the app on a live server

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,188 @@
...
+### Install
+
+Install the required packages `npm install`
+
+### Usage
+
+To run the project use `npm run start`
+To fully interact with the backend check this [repo](https://github.com/M-AminAlizadeh/Booking-doctor-appointment-back-end)
+
+### Run tests
```
Comment (Meltrust) [inline README.md:105]: - [ ] There are 3 backend tests that are not passing:

image

Kindly correct them. 👍

Comment (Meltrust) [inline README.md:105]: - [ ]  You are using Rspec, but the "test" folder still exists, which is used by another test framework. Kindly consider deleting that folder if you are not using it. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!










#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Readme has been improved✔️




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Comment by SaveryIV** [comment]
@Meltrust I fixed the tests, and actually delete the folder test

File: README.md [hunk 2]
```diff
@@ -1 +1,198 @@
...
+Execute all the migrations for testing `rails/rake db:migrate RAILS_ENV=test`
+
+### Usage
+
+To run the project use `rails s`
+To fully interact with the backend check this [repo](https://github.com/M-AminAlizadeh/Booking-doctor-appointment-back-end)
+
+### Run tests
+
+To run tests use `rspec spec`
```
Comment (Meltrust) [inline README.md:111]: - [ ] OPTIONAL: One test is still not passing.  I would advise you to correct it before final evaluation:

image

image

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**

## Thank you for the changes implemented 💪 🥇 ㊗️ 



### To highlight

- Most tests have been corrected ✔️ 



Cheers and Happy coding!👏👏👏


------

File: README.md (hunk 2 above)
Comment (SaveryIV) [reply README.md:111]: In my case is working but i will check

**Pull request merged**